storagedriver/azure/azure.go

```diff
@@ -272,6 +361,24 @@ func (d *Driver) Delete(path string) error {
 	return nil
 }
 
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *Driver) URLFor(path string, options map[string]interface{}) (string, error) {
```

Comment:
Will the URL provided here be usable by clients without extra authentication provisions? How does this work for azure? Is it similar to s3 signed url?

Cool to see this in there.